types/src/deploy/approvals_hash.rs

```diff
@@ -1,53 +1,51 @@
-use std::{
-    collections::BTreeSet,
-    fmt::{self, Debug, Display, Formatter},
-};
+use alloc::{collections::BTreeSet, vec::Vec};
+use core::fmt::{self, Display, Formatter};
 
+#[cfg(feature = "datasize")]
 use datasize::DataSize;
 #[cfg(any(feature = "testing", test))]
 use rand::Rng;
 use serde::{Deserialize, Serialize};
 
+use super::Approval;
 #[cfg(any(feature = "testing", test))]
-use casper_types::testing::TestRng;
-use casper_types::{
+use crate::testing::TestRng;
+use crate::{
     bytesrepr::{self, FromBytes, ToBytes},
     Digest,
 };
 
-use super::Approval;
-
 /// The cryptographic hash of the bytesrepr-encoded set of approvals for a single deploy.
 #[derive(
-    Copy,
-    Clone,
-    DataSize,
-    Ord,
-    PartialOrd,
-    Eq,
-    PartialEq,
-    Hash,
-    Serialize,
-    Deserialize,
-    Debug,
-    Default,
+    Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Serialize, Deserialize, Debug, Default,
 )]
+#[cfg_attr(feature = "datasize", derive(DataSize))]
 #[serde(deny_unknown_fields)]
 pub struct ApprovalsHash(Digest);
 
 impl ApprovalsHash {
+    /// The number of bytes in a `ApprovalsHash` digest.
+    pub const LENGTH: usize = Digest::LENGTH;
+
     /// Constructs a new `ApprovalsHash` by bytesrepr-encoding `approvals` and creating a [`Digest`]
     /// of this.
     pub fn compute(approvals: &BTreeSet<Approval>) -> Result<Self, bytesrepr::Error> {
         let digest = Digest::hash(approvals.to_bytes()?);
         Ok(ApprovalsHash(digest))
     }
 
-    /// Returns the wrapped inner hash digest.
+    /// Returns the wrapped inner digest.
```

Comment:
separate ticket, but I think we should consider making specific newtypes for all of these various naked Digests which are not really interchangeable, like we discussed for ChainspecHash.